app/models/collection.rb

```diff
@@ -162,6 +162,54 @@ def has_untranscribed_pages?
     self.works.joins(:work_statistic).where('work_statistics.total_pages > work_statistics.transcribed_pages').exists?
   end
 
+  def update_works_stats
+    works = self.works.includes(:work_statistic)
+    works_stats = get_works_stats_hash
+    works.each do |w|
+      w.work_statistic.recalculate_from_hash(works_stats[w.id]) if works_stats.key?(w.id)
+    end
+    calculate_complete
+  end
+
+  def enable_ocr
+    works.update_all(ocr_correction: true)
+    update_works_stats
+  end
+
+  def disable_ocr
+    works.update_all(ocr_correction: false)
+    update_works_stats
+  end
+
+  def get_works_stats_hash
+    stats = {}
+    work_prototype = {
+      transcription: {},
+      translation: {},
+      total: 0
+    }
+    work_ids = works.ids
+    transcription = Page.where(work_id: work_ids).group(:work_id, :status).count
+    translation = Page.where(work_id: work_ids).group(:work_id, :translation_status).count
+    totals = Page.where(work_id: work_ids).group(:work_id).count
+    
+    transcription.each do |(id, status), value|
+      stats[id] = work_prototype unless stats.has_key?(id)
```

Comment:
I'm not sure why I dislike testing on `unless stats.has_key?(id)` instead of `unless stats[id]` or `if stats[id].nil?`

A shorter alternative is `stats[id] ||= work_prototype`, although I've sometimes found that hard to read.

Reply:
opted for `if stats[id].nil?`, because I think it is nice and readable and communicates the intent of the code well.